router.go

```
@@ -79,6 +84,54 @@ func setupRouter() (r *gin.Engine, close func()) {
 	return
 }
 
+/*  the function search for the authrization header to check if the client has a jwt token.
+	If the token is not valid or expired, it will redirect the client to the auth service
+	If the token is valid, it will inject user to the gin context
+*/
 func authRequired(c *gin.Context) {
-	c.Set("User", user{id: "testuser"})
+	auth := c.GetHeader("Authorization")
+	if auth == "" {
+		redirectToAuthService(c)
+		return
+	}
+	authArr := strings.Fields(auth)
+	_, tokenString := authArr[0], authArr[1]
+	
+	secret := viper.GetString(configSecret)
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		// Validates the alg is what we expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
```

You are formatting an error with `fmt.Errorf` but you are not doing anything with its returned error.
Also errors should start with a lowercase character.
You should return an error in this anonymous function and handle `jwt.Parse` error by calling `redirectToAuthService` instead of calling it in line 106.

---

+1
I agree with your first comment,
about the second, we discussed about it and decided to let the user to authenticate again instead of return an error.

but I do think you should return after line 106